[DTensor] implement aten.equal sharding prop

XilunWu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95852
* __->__ #97170

XilunWu committed:
[DTensor] implement aten.equal sharding prop

[ghstack-poisoned]

XilunWu committed:
Update on "[DTensor] implement aten.equal sharding prop"




[ghstack-poisoned]

wanchaol reviewed (approved):
lgtm, thanks for the fix!

wanchaol commented on test/distributed/_tensor/test_tensor_ops.py:
```diff
@@ -234,6 +234,27 @@ def test_zeros_like(self):
+        dist_tensor_1 = DTensor.from_local(input_tensor_1, device_mesh, shard_spec)
+        dist_tensor_2 = DTensor.from_local(input_tensor_2, device_mesh, shard_spec)
+        # TODO: equal op currently returns each shard's local result
+        eq_result = dist_tensor_1.equal(dist_tensor_2)
```
maybe this should be under a raise Exception condition? It's wrong now, so we would fix it later by removing that exception

XilunWu committed:
Update on "[DTensor] implement aten.equal sharding prop"




[ghstack-poisoned]

XilunWu committed:
Update on "[DTensor] implement aten.equal sharding prop"




[ghstack-poisoned]

XilunWu commented:
@XilunWu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44321773).

XilunWu committed:
Update on "[DTensor] implement aten.equal sharding prop"


Differential Revision: [D44321773](https://our.internmc.facebook.com/intern/diff/D44321773)

[ghstack-poisoned]

XilunWu commented:
@XilunWu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44321773).
